controllers/seedgen_controller.go

```diff
@@ -646,6 +679,17 @@ func (r *SeedGeneratorReconciler) generateSeedImage(ctx context.Context, seedgen
 		r.Log.Info(fmt.Sprintf("No hubKubeconfig found in secret %s. Skipping hub interaction", utils.SeedGenSecretName))
 	}
 
+	// Get the cluster's pull-secret
+	clusterPullSecret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: common.PullSecretName, Namespace: common.OpenshiftConfigNamespace}, clusterPullSecret); err != nil {
```

Comment:
We have pretty nice function in util to GetSecretData i think , better to use it

Reply:
added #292 to align with this function